compiler-rt/lib/asan/asan_cheerpwasm.cpp

```diff
@@ -0,0 +1,94 @@
+#include "asan_interceptors.h"
+#include "asan_internal.h"
+#include "asan_mapping.h"
+#include "asan_poisoning.h"
+#include "asan_stack.h"
+#include "asan_thread.h"
+
+#if SANITIZER_CHEERPWASM
+#include "lsan/lsan_common.h"
+#include <stdlib.h>
+
+namespace __asan {
+
+uptr FindDynamicShadowStart() {
+  uptr shadow_size_bytes = MemToShadowSize(kHighMemEnd);
+  return MapDynamicShadow(shadow_size_bytes, ASAN_SHADOW_SCALE,
+                          /*min_shadow_base_alignment*/ 0, kHighMemEnd);
+}
+
+void AsanCheckDynamicRTPrereqs() {}
+void AsanCheckIncompatibleRT() {}
+void InitializePlatformInterceptors() {}
+void InitializePlatformExceptionHandlers() {}
+bool IsSystemHeapAddress (uptr addr) { return false; }
+void ReplaceSystemMalloc() {}
+bool PlatformUnpoisonStacks() { return false; }
+
+void *AsanDoesNotSupportStaticLinkage() {
+  // On Linux, this is some magic that fails linking with -static.
+  // On Cheerp, we have to do static linking, so we stub this out.
+  return nullptr;
+}
+
+void InitializeAsanInterceptors() {}
+
+void FlushUnneededASanShadowMemory(uptr p, uptr size) {}
+
+static void (*tsd_destructor)(void *tsd) = nullptr;
+
+struct tsd_key {
+  // tsd_key() : key(nullptr) {} can't use this, since it will try to initialize
```

Comment:
Hmmm, I remember explicitly commenting this out, but re-running some of my tests shows that it might not be the case anymore. I'll do some more thorough testing and come back to you on this